### DISCUSSION TITLE
Quadratic constraints contain large coefficients on linear part

### DISCUSSION THREAD

**Discussion by wendy444444** [yalmip/YALMIP#1127] [General discussion] [answered]
I use matlab+yalmip+gurobi to solve a mixed integer nonlinear programming problem. The parameters I set are as follows:
`options = sdpsettings('verbose',2,'solver','GUROBI', 'debug', 1, 'gurobi.NonConvex', 2,'gurobi.timelimit',300,'gurobi.MIPGap', 10^(-4),'gurobi.NumericFocus',3);%cplex
`
Then the results are as follows:
```
Set parameter Username
Set parameter TimeLimit to value 300
Set parameter CrossoverBasis to value 0
Set parameter NodefileDir to value ""
Set parameter NonConvex to value 2
Set parameter NumericFocus to value 3
Set parameter PreSOS2BigM to value 0
Set parameter TuneTrials to value 3
Academic license - for non-commercial use only - expires 2022-06-11
Gurobi Optimizer version 9.5.1 build v9.5.1rc2 (win64)
Thread count: 2 physical cores, 4 logical processors, using up to 4 threads
Optimize a model with 21417 rows, 8045 columns and 41884 nonzeros
Model fingerprint: 0x6159bf80
Model has 4110 quadratic constraints
Variable types: 3840 continuous, 4205 integer (4205 binary)
Coefficient statistics:
  Matrix range     [1e-05, 7e+04]
  QMatrix range    [8e-01, 1e+01]
  QLMatrix range   [1e+00, 1e+09]
  Objective range  [1e+00, 1e+00]
  Bounds range     [1e+00, 1e+00]
  RHS range        [1e-05, 1e+07]
  QRHS range       [1e+09, 1e+09]
Warning: Quadratic constraints contain large coefficients on linear part
Presolve removed 19646 rows and 7060 columns
Presolve time: 1.23s
Presolved: 4010 rows, 1338 columns, 10522 nonzeros
Presolved model has 56 quadratic constraint(s)
Presolved model has 607 bilinear constraint(s)
Variable types: 1023 continuous, 315 integer (301 binary)

Root relaxation: objective -2.155180e+07, 596 iterations, 0.05 seconds (0.01 work units)

    Nodes    |    Current Node    |     Objective Bounds      |     Work
 Expl Unexpl |  Obj  Depth IntInf | Incumbent    BestBd   Gap | It/Node Time

     0     0 -2.155e+07    0  203          - -2.155e+07      -     -    1s
H    0     0                      -0.0000001 -2.098e+07      -     -    2s
     0     2 -2.098e+07    0  203   -0.00000 -2.098e+07      -     -    2s
   152    74 infeasible   20        -0.00000 -1.922e+07      -  66.7    5s
   929   255 -1.747e+07   14   30   -0.00000 -1.919e+07      -  43.7   10s

Explored 935 nodes (42030 simplex iterations) in 10.60 seconds (3.82 work units)
Thread count was 4 (of 4 available processors)

Solution count 2: -1.08022e-07 -1.08022e-07 

Optimal solution found (tolerance 1.00e-04)
Best objective -1.080215916716e-07, best bound -1.080215916716e-07, gap 0.0000%
Elapsed time is 145.198765 seconds.
```
My questions are as follows: 
1. I didn't set the limit of the objective function, but the result was Objective range  [1e+00, 1e+00]. I don't understand why this happened, and I don't know if it will affect the result. If I want to modify it, how can I modify it?
2. The quadratic constraints I set are as follows:
V=sdpvar(11,20,'full');
V_12=sdpvar(11,20,'full');
V22=[12*ones(1,20);V_12];
0<=V,V<=1.1*12
0<=V22,V22<=1.1*1.1*12*12;
V22(i,t)==V(i,t)*V(i,t);
How can I correct the warning in the results?
3. I got a value of 10 (-7) which is close to 0 but not 0, so I want to make the accuracy of the obtained result larger. If I want to set the accuracy of the result, how should I set it?

**Comment by johanlofberg** [comment, marked as answer]
1. the non-zero coefficeints in the objective all  have absolute value 1 
2. 2. you simply have to study your constraints and see where you are using these problematically large coefficients, and see what you can do, by e.g. scaling the problem or what ever
3. 3. 10^-7 is at the level of solver precision. Sure, look for and tweak some solver option, but normally that makes no difference. concentrate on creating good models, not desperately tweaking solver settings

**Reply by wendy444444** [reply]
Thank you, sir. After modifying some data, the program can run, but after running, a warning appears and some constraints are ignored.
`Warning: max constraint violation (9.9000e-06) exceeds tolerance` 
What should I do?

**Reply by johanlofberg** [reply]
It simply means that the problem most likely is badly scaled as solver tolerance was not met after having solved the problem and scaling back to original variables etc, or the problem is simply ill-posed and is marginally infeasible. You either work on fixing that, or you accept that some constraints are violated with 10^-5

**Reply by wendy444444** [reply]
Then if I want to solve the problem, should I modify the constraint or the data?

**Reply by johanlofberg** [reply]
That's what you have to figure out. We cannot help you anymore. You have been given the same answer many times. If you have problems with numerics, you have to improve the model. How to improve the model is impossible for anyone to know or figure out except you.
